services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/tenant/TenantInformationService.java

```diff
@@ -42,4 +44,24 @@
      */
      Future<Result<TenantKey>> tenantExists(String tenantId, Span span);
 
+    /**
+     * Gets the tenant identified by the given id.
+     *
+     * @param tenantId The id identifying the tenant to obtain, must not be {@code null}.
+     * @param span The active OpenTracing span for this operation, must not be {@code null}.
+     *            It is not to be closed in this method! An implementation should log (error) events on this span and
+     *            it may set tags and use this span as the parent for any spans created in this method.
+     *
+     * @return A future indicating the outcome of the operation.
+     *             The <em>status</em> will be
+     *             <ul>
+     *             <li><em>200 OK</em> if a tenant with the given ID is registered.
+     *             The <em>payload</em> will contain the tenant's configuration information.</li>
+     *             <li><em>404 Not Found</em> if no tenant with the given identifier exists.</li>
+     *             </ul>
```

Comment:
No, as far as I can see not, in an error case it will always be a TenantResult with status code 404, similar to _tenantExists_.

Reply:
Then the wording should probably changed to reflect the fact that this will always return a succeeded future?

Reply:
I think this method should be in line with the `TenantInformationService.tenantExists` method and the `TenantService` methods in that respect.
These other methods don't make the promise, that always a succeeded Future will be returned. And looking at the current implementation of `TenantService.get` in the JDBC implementation, the returned Future can actually be failed. This would mean that `AutowiredTenantInformationService.getTenant` would also return a failed Future.
Therefore I would keep the javadoc as is, and maybe change all related method javadocs and the JDBC implementation in that respect in a separate PR.